application/ui/components/params.jsx

```diff
@@ -0,0 +1,170 @@
+    {
+        var values   = {};
+        var includes = {};
+
+        _.each(this.props.params, function(param) {
+            includes[param.name] = true;
+
+            if (typeof param.defaultValue !== 'undefined') {
+                values[param.name] = param.defaultValue;
+                return;
+            }
+
+            switch (param.type) {
+                case 'string':
+                    values[param.name] = '';
+                    break;
+                case 'number':
+                    values[param.name] = 0;
+                    break;
+                case 'boolean':
+                    values[param.name] = false;
+                    break;
+                case 'enum':
+                    values[param.name] = param.defaultValue;
+                    break;
+                case 'array':
+                    values[param.name] = [];
+                    break;
+                case 'object':
+                case 'hash':
+                case 'custom-object':
+                    values[param.name] = {};
+                    break;
+                case 'stripe-token':
+                    values[param.name] = null;
+                    break;
+            }
+        });
+
+        return {
+            values      : values,
+            nullKeys    : {},
+            includeKeys : includes
+        };
+    },
+
+    getValue : function()
+    {
+        var data = {}, paramName;
+
+        for (var i in this.props.params) {
+            paramName = this.props.params[i].name;
+            if (typeof this.state.includeKeys[paramName] !== 'undefined') {
+                if (typeof this.state.nullKeys[paramName] === 'undefined') {
+                    data[paramName] = this.state.values[paramName];
+                } else {
+                    data[paramName] = null;
+                }
+            }
+        }
+
+        return data;
+    },
+
+    /**
+     * Manage null values
+     * @param key
+     * @param event
+     */
+    nullHandler : function(key, event)
+    {
+        var state = _.extend({}, this.state);
+        var value = event.currentTarget.checked;
+
+        if (value) {
+            state.nullKeys[key] = true;
+        } else {
+            delete state.nullKeys[key];
+        }
+
+        this.props.onChange(this.getValue());
+        this.setState(state);
+    },
+
+    includeHandler : function(key, event)
+    {
+        var state = _.extend({}, this.state);
+        var value = event.currentTarget.checked;
+
+        if (value) {
+            state.includeKeys[key] = true;
+        } else {
+            delete state.includeKeys[key];
+        }
+
+        this.setState(state);
+        this.props.onChange(this.getValue());
+    },
+
+    changeHandler : function(key, value)
+    {
+        var state = _.extend({}, this.state);
+
+        state.values[key] = value;
+
+        this.setState(state, function() {
+            this.props.onChange(this.getValue());
+        });
```

Comment:
Would it be more elegant to make the `Params` components "controlled" components? i.e. Remove `values` from state, and just let it be passed in as a prop?

As it currently stands, every layer of nested `Param` components (as well as the parent of the top-level Param) hold copies of the same data. (Although at different layers.) So the benefit would be reducing this to a single copy of `values`, preventing potential bugs where the different copies of `values` get out of sync.

This would be no small change, so even if it is a desirable change, perhaps it belongs in a separate Refactor issue.